STACpopulator/stac_utils.py

```diff
@@ -47,395 +31,127 @@ def url_validate(target: str) -> bool:
-               "opendap_service": pystac.MediaType.HTML,
-               "wcs_service": pystac.MediaType.XML,
-               "wms_service": pystac.MediaType.XML,
-               "nccs_service": "application/x-netcdf",
-               "HTTPServer": "application/x-netcdf",
-               "OPENDAP": pystac.MediaType.HTML,
-               "NCML": pystac.MediaType.XML,
-               "WCS": pystac.MediaType.XML,
-               "ISO": pystac.MediaType.XML,
-               "WMS": pystac.MediaType.XML,
-               "NetcdfSubset": "application/x-netcdf",
-               }
+    cfmeta = attrs["groups"]["CFMetadata"]["attributes"]
+
+    # Create pydantic STAC item
+    item = STACItem(
+        id=iid,
+        geometry=ncattrs_to_geometry(attrs),
+        bbox=ncattrs_to_bbox(attrs),
+        properties=item_props_datamodel(
+            start_datetime=cfmeta["time_coverage_start"],
+            end_datetime=cfmeta["time_coverage_end"],
+            **attrs["attributes"],
+        ),
+        datetime=None,
+    )
 
-asset_roles = {"httpserver_service": ["data"],
-               "opendap_service": ["data"],
-               "wcs_service": ["data"],
-               "wms_service": ["visual"],
-               "nccs_service": ["data"],
-               "HTTPServer": ["data"],
-               "OPENDAP": ["data"],
-               "NCML": ["metadata"],
-               "WCS": ["data"],
-               "ISO": ["metadata"],
-               "WMS": ["visual"],
-               "NetcdfSubset": ["data"],}
+    # Convert pydantic STAC item to a PySTAC Item
+    item = pystac.Item(**json.loads(item.model_dump_json(by_alias=True)))
+
+    # Add assets
+    if "access_urls" in attrs:
+        root = attrs["access_urls"]
+    elif "THREDDSMetadata" in attrs["groups"]:
+        root = attrs["groups"]["THREDDSMetadata"]["groups"]["services"]["attributes"]
+    else:
+        root = {}
+
+    for name, url in root.items():
+        asset = pystac.Asset(href=url, media_type=media_types.get(name), roles=asset_roles.get(name))
+        item.add_asset(name, asset)
+
+    return item
+
+
+media_types = {
+    "httpserver_service": "application/x-netcdf",
+    "opendap_service": pystac.MediaType.HTML,
+    "wcs_service": pystac.MediaType.XML,
+    "wms_service": pystac.MediaType.XML,
+    "nccs_service": "application/x-netcdf",
+    "HTTPServer": "application/x-netcdf",
+    "OPENDAP": pystac.MediaType.HTML,
+    "NCML": pystac.MediaType.XML,
+    "WCS": pystac.MediaType.XML,
+    "ISO": pystac.MediaType.XML,
+    "WMS": pystac.MediaType.XML,
+    "NetcdfSubset": "application/x-netcdf",
+}
+
+asset_roles = {
+    "httpserver_service": ["data"],
+    "opendap_service": ["data"],
+    "wcs_service": ["data"],
+    "wms_service": ["visual"],
+    "nccs_service": ["data"],
+    "HTTPServer": ["data"],
+    "OPENDAP": ["data"],
+    "NCML": ["metadata"],
+    "WCS": ["data"],
+    "ISO": ["metadata"],
+    "WMS": ["visual"],
+    "NetcdfSubset": ["data"],
+}
```

Comment:
These look like THREDDS-specific metadata. They should most probably be attributes within its implementation.